insights/parsers/grubby.py

```diff
@@ -2,29 +2,166 @@
+from insights import CommandParser, LegacyItemAccess
+from insights import parser
 
 
 @parser(Specs.grubby_default_index)
-class GrubbyDefaultIndex(Parser):
+class GrubbyDefaultIndex(CommandParser):
     """
-    This parser will parse the output of command ``grubby --default-index``.
+    This parser parses the output of command ``grubby --default-index``.
+
+    The typical output of this command is::
+
+        0
+
+    Examples:
+        >>> grubby_default_index.default_index
+        0
+
+    Raises:
+        SkipException: When output is empty
+        ParseException: When output is invalid
 
     Attributes:
         default_index (int): the numeric index of the current default boot entry, count from 0
     """
     def parse_content(self, content):
-        if content and content[0].isdigit():
-            self.default_index = int(content[0])
-        else:
-            raise SkipException('Invalid default index value.')
+        if not content:
+            raise SkipException('Empty output')
+        if len(content) != 1 or not content[0].isdigit():
+            raise ParseException('Invalid output: {0}', content)
+        self.default_index = int(content[0])
+
+
+@parser(Specs.grubby_default_kernel)
+class GrubbyDefaultKernel(CommandParser):
+    """
+    This parser parses the output of command ``grubby --default-kernel``.
+
+    The typical output of this command is::
+
+        /boot/vmlinuz-2.6.32-573.el6.x86_64
+
+    Examples:
+
+        >>> grubby_default_kernel.default_kernel
+        '/boot/vmlinuz-2.6.32-573.el6.x86_64'
+
+    Raises:
+        SkipException: When output is empty
+        ParseException: When output is invalid
+
+    Attributes:
+        default_kernel(str): The default kernel name for next boot
+    """
+    def parse_content(self, content):
+        if not content:
+            raise SkipException('Empty output')
+        if len(content) != 1:
+            raise ParseException('Invalid output: {0}', content)
+        self.default_kernel = content[0].strip()
+
+
+@parser(Specs.grubby_info_all)
+class GrubbyInfoALL(CommandParser, LegacyItemAccess):
+    """
+    This parser parses the output of command ``grubby --info=ALL``.
+
+    The typical output of this command is::
+
+        boot=/dev/vda
+        index=0
+        kernel=/vmlinuz-2.6.32-754.9.1.el6.x86_64
+        args="ro rd_NO_LUKS rd_NO_LVM LANG=en_US.UTF-8 rd_NO_MD SYSFONT=latarcyrheb-sun16 crashkernel=auto  KEYBOARDTYPE=pc KEYTABLE=us rd_NO_DM rhgb quiet"
+        root=UUID=1b46779e-4fae-442d-a2ac-dd6d8563ff3e
+        initrd=/boot/initramfs-2.6.32-754.9.1.el6.x86_64.img
+        index=1
+        kernel=/vmlinuz-2.6.32-573.el6.x86_64
+        args="ro rd_NO_LUKS rd_NO_LVM LANG=en_US.UTF-8 rd_NO_MD SYSFONT=latarcyrheb-sun16 crashkernel=auto  KEYBOARDTYPE=pc KEYTABLE=us rd_NO_DM rhgb quiet"
+        root=UUID=1b46779e-4fae-442d-a2ac-dd6d8563ff3e
+        initrd=/boot/initramfs-2.6.32-573.el6.x86_64.img
```

Comment:
Please note that the kernel can be reused multiple times in different entry, for example:

```
boot=/dev/sda
index=0
kernel=/vmlinuz-2.6.32-431.el6.x86_64
args="ro rd_NO_LUKS LANG=en_US.UTF-8 rd_NO_MD rd_LVM_LV=VolGroup/lv_swap SYSFONT=latarcyrheb-sun16 crashkernel=auto rd_LVM_LV=VolGroup/lv_root  KEYBOARDTYPE=pc KEYTABLE=us rd_NO_DM"
root=/dev/mapper/VolGroup-lv_root
initrd=/boot/initramfs-2.6.32-431.el6.x86_64.img
index=1
kernel=/vmlinuz-2.6.32-431.el6.x86_64
args="ro rd_NO_LUKS LANG=en_US.UTF-8 rd_NO_MD rd_LVM_LV=VolGroup/lv_swap SYSFONT=latarcyrheb-sun16 crashkernel=auto rd_LVM_LV=VolGroup/lv_root  KEYBOARDTYPE=pc KEYTABLE=us rd_NO_DM quiet"
root=/dev/mapper/VolGroup-lv_root
initrd=/boot/initramfs-2.6.32-431.el6.x86_64.img
```
In this case we may need to rethink the `__getitem__()` method.

Reply:
The `__getitem__` is just used to get specified kernel entry, but not the booting entry.
To get the next booting entry, need to corporate with the GrubbyDefaultKernel which indicates the next booting kernel.

Reply:
Oh, got your point

Reply:
@lonicerae - Thanks for reviewing. I removed the GrubInfoALL parser for now, please check it again.